### PR TITLE
Fixes #1678 fix LimitFields group commonality bug

### LINKED ISSUE #1678: LimitFields hit term commonality check bug
https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java#L183 compares groupings with .endsWith instead of .equals causing us to return extra fields.  We will return group 8 if 18 is in the HIT_TERMS, 2 if 12, 3 if 33, etc.

### PR DESCRIPTION
**Pull request by hlgp**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java [hunk 1]
```diff
@@ -180,7 +180,8 @@ private void manageHitsAndMisses(String keyWithGrouping, String keyNoGrouping, A
                         String[] commonalityAndGroupingContext = LimitFields.getCommonalityAndGroupingContext(key);
                         if (commonalityAndGroupingContext != null) {
                             String hitTermKeyCommonality = commonalityAndGroupingContext[0];
-                            if (hitTermKeyCommonality.equals(keyWithGroupingCommonality) && key.endsWith(keyWithGroupingSuffix)) {
+                            String hitTermGroup = commonalityAndGroupingContext[1];
+                            if (hitTermKeyCommonality.equals(keyWithGroupingCommonality) && keyWithGroupingSuffix.equals(hitTermGroup)) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java:184]: Can we add an issue to refactor and clean up some of the extra object creation here?

**Review by apmoriarty — APPROVED** [review]

File: warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java (hunk 1 above)
Comment (hlgp) [reply warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java:184]: feel free to open one

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java:184]: Opened https://github.com/NationalSecurityAgency/datawave/issues/1680

**Review by ivakegg — APPROVED** [review]
Nice job!

**Pull request merged**